proposal.md

```diff
@@ -0,0 +1,186 @@
+# Module Interpretation in Node.js Proposal
+
+## Introduction
+
+There are two general areas where Node loads JavaScript and therefore needs a way to know whether to parse the JavaScript source code as CommonJS or as ESM:
+
+1. Initial entry points, such as running `node file.js` on the command line, or running `node` with  `--eval`, piping source code to Node via `STDIN`, or running an extensionless executable JavaScript file with a hashbang.
+
+2. `import` statements and `import()` expressions, where a file or package is specified as the source to be imported, e.g. `import _ from 'lodash'`.
+
+This proposal aims to cover how Node should decide what module system to use for each of the use cases within the above groups. This is the _interpretation_ part of module resolution; the other part is _location._ Locating the code to be interpreted is covered by the [resolver specification][nodejs/ecmascript-modules:esm.md#resolver-algorithm], and involves things like taking the specifier string `'underscore'` and finding its package entry point file `./node_modules/underscore/underscore.js`. Location is out of scope for this proposal. This proposal covers what Node should do once the source code to be parsed has already been located.
+
+A central new concept used by many of the methods suggested in the proposal is the _package scope,_ delineated by _package boundaries._
+
+## Package scopes and package boundaries
```

Comment:
```suggestion
## Package boundaries and scopes
```

First there is the boundary, then there is the scope, until the next boundary… etc. No chickens here.